src/inc/til/bitmap.h

```diff
@@ -144,28 +182,137 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
             return const_iterator(_bits, _sz, _sz.area());
         }
 
+        const std::vector<til::rectangle>& runs() const
+        {
+            // If we don't have cached runs, rebuild.
+            if (!_runs.has_value())
+            {
+                // If there's only one square dirty, quick save it off and be done.
+                if (one())
+                {
+                    // We're const for the sake of actually manipulating the dirty state.
+                    // But we remove const for updating the cache.
+                    _runs.emplace({ _dirty.value() });
+                }
+                else
+                {
+                    // We're const for the sake of actually manipulating the dirty state.
+                    // But we remove const for updating the cache.

```

Comment:
const cast is gone, so comments don't mean anything

Reply:
woops, yeah, thanks.